File: kotlin/1334.kt
class Solution {
    fun findTheCity(n: Int, edges: Array<IntArray>, distanceThreshold: Int): Int {
        val INF = 1_000_000_000
        val dist = Array(n) { IntArray(n) { INF } }

        for (i in 0 until n) dist[i][i] = 0

        for ((u, v, w) in edges) {
            dist[u][v] = w
            dist[v][u] = w
        }

        for (k in 0 until n) {
            for (i in 0 until n) {
                if (dist[i][k] == INF) continue
                for (j in 0 until n) {
                    if (dist[k][j] != INF && dist[i][k] + dist[k][j] < dist[i][j]) {
                        dist[i][j] = dist[i][k] + dist[k][j]
                    }
                }
            }
        }

        var minReachable = n
        var resultCity = -1

        for (i in 0 until n) {
            var count = 0
            for (j in 0 until n) {
                if (i != j && dist[i][j] <= distanceThreshold) {
                    count++
                }
            }
            if (count <= minReachable) {
                minReachable = count
                resultCity = i
            }
        }

        return resultCity
    }
}

File: solutions/1334.py
import heapq

def findTheCity(n: int, edges: list[list[int]], distanceThreshold: int) -> int:
    graph = [[] for _ in range(n)]
    for u, v, w in edges:
        graph[u].append((v, w))
        graph[v].append((u, w))

    def get_reachable_count(start: int) -> int:
        dist = [float('inf')] * n
        dist[start] = 0
        pq = [(0, start)]
        
        while pq:
            d, u = heapq.heappop(pq)
            if d > dist[u]:
                continue
                
            for v, weight in graph[u]:
                new_d = d + weight
                if new_d <= distanceThreshold and new_d < dist[v]:
                    dist[v] = new_d
                    heapq.heappush(pq, (new_d, v))
        
        return sum(1 for d in dist if d <= distanceThreshold) - 1

    min_reachable = float('inf')
    best_city = -1
    
    for i in range(n):
        reachable = get_reachable_count(i)
        if reachable <= min_reachable:
            min_reachable = reachable
            best_city = i
            
    return best_city


